client/websocket_client.jsx

```diff
@@ -201,4 +211,22 @@ export default class WebSocketClient {
         data.user_ids = userIds;
         this.sendMessage('get_statuses_by_ids', data, callback);
     }
+
+    createPingEvent() {
+        this.pingTimer = setInterval(() => {
+            this.sendMessage('ping');
```

Comment:
there is the option [here](https://github.com/mattermost/mattermost-webapp/blob/f4f051820cc89e66654803cbb36980a8be849a8c/client/websocket_client.jsx#L183) of not sending the message. There are also some exceptions (see [MDN](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/send)) we are not managing.

if any of the above cases happen, we shouldn't wait for pong and call close (or try to restore) inmediately.

Reply:
Sorry, I didn't get it. This is for closing the socket in the event when socket. does not close by itself. Exceptions are for send on the socket and not related to close. 

> there is the option here of not sending the message.

If socket conn is closed i.e this.conn === null;  then we don't emit ping messages so it does not matter all that much